File: backend/internal/model/share.go
package model

// Share 分享模型
type Share struct {
	ID           string `gorm:"primaryKey;type:varchar(50)" json:"id"`
	ResourceType string `gorm:"type:varchar(50);not null" json:"resourceType"` // dashboard, chart
	ResourceID   string `gorm:"type:varchar(50);not null" json:"resourceId"`
	Token        string `gorm:"type:varchar(50);uniqueIndex" json:"token"`
	Password     string `gorm:"type:varchar(100)" json:"password,omitempty"`
	ExpireTime   int64  `gorm:"default:0" json:"expireTime"` // 过期时间,0表示永不过期
	CreateTime   int64  `gorm:"autoCreateTime:milli" json:"createTime"`
	CreateBy     string `gorm:"type:varchar(50)" json:"createBy"`
}

func (Share) TableName() string {
	return "sys_share"
}

File: Archive/frontend/src/api/share.ts
import request from '../utils/request';

export interface Share {
    id: string;
    resourceType: string;
    resourceId: string;
    token: string;
    password?: string;
    expireTime: number;
    createTime: number;
    createBy: string;
}

export const shareAPI = {
    create: (data: Partial<Share>) => request.post<Share>('/share', data),
    get: (id: string) => request.get<Share>(`/share/${id}`),
    delete: (id: string) => request.delete(`/share/${id}`),
    list: (resourceType: string, resourceId: string) =>
        request.get<Share[]>(`/share?resourceType=${resourceType}&resourceId=${resourceId}`),
    validate: (token: string, password?: string) =>
        request.get<Share>(`/share/validate/${token}${password ? `?password=${password}` : ''}`),
};
